mapusaurus/hmda/views.py

```diff
@@ -28,38 +28,50 @@ def loan_originations(request):
     query = HMDARecord.objects.filter(base_hmda_query())
     if action_taken_selected:
         query = query.filter(action_taken__in=action_taken_selected)
-    if institution_selected:
-        if lender_hierarchy == 'true':
-            hierarchy_list = institution_selected.get_lender_hierarchy(False, False)
-            if len(hierarchy_list) > 0:
-                query = query.filter(institution__in=hierarchy_list) 
+    #if lender param key is passed in
+    if institution_id:
+        #get institution_selected
+        institution_selected = Institution.objects.filter(pk=institution_id).first()
+        #if param value is valid
+        if institution_selected: 
```

Comment:
Ah, gotcha. I got confused. It might be worth calling this out explicitly, so the logic is more visible to help readers:
```python
if institution_id and not institution_selected:
    return None
elif institution_selected:
    query = add_hierarchy_filter(query, ...)
    query = add_peer_filter(query, ...)
```